integrationtests/git/git_test.go

```diff
@@ -0,0 +1,374 @@
+package integrationtests
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"os"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/go-git/go-git/v5/plumbing/transport"
+	cp "github.com/otiai10/copy"
+	"github.com/rancher/gitjob/integrationtests/git/util"
+	gitjobv1 "github.com/rancher/gitjob/pkg/apis/gitjob.cattle.io/v1"
+	"github.com/rancher/gitjob/pkg/git"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+	"golang.org/x/crypto/ssh"
+	v1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+/*
+This tests uses gogs for testing the integration with a git server. Gogs container uses the data from assets/gitserver, which
+contains one user, one public repository, and another private repository. Initial commits and fingerprint are provided as consts.
+*/
+const (
+	latestCommitPublicRepo  = "8cd5ab9c851482ce13a544c91ee010f6fdc7cf3f"
+	latestCommitPrivateRepo = "417310891d63d3f3a478bd4c5013e2f532056e8e"
+	gogsFingerPrint         = "ecdsa-sha2-nistp256 AAAAE2VjZHNhLXNoYTItbmlzdHAyNTYAAAAIbmlzdHAyNTYAAABBBBpayjxZ7oeeMc6KjGM0VgFEE5GmN1H6RLquUENLcpGcKzrEtym48WmAnX9Xwdkg8eMUBgyYkZtZgR+eapf29fQ="
+	gogsUser                = "test"
+	gogsPass                = "pass"
+)
+
+func TestLatestCommit_NoAuth(t *testing.T) {
+	ctx := context.Background()
+	container, url, err := createGogsContainer(ctx, createTempFolder(t))
+	if err != nil {
+		t.Errorf("got error when none was expected: %v", err)
+	}
```

Comment:
nit: using https://pkg.go.dev/github.com/stretchr/testify/require, this could simply be:
```suggestion
	require.NoError(t, err)
```
(Same remark about `if` blocks throughout this file, checking expectations vs reality, with `stretchr/testify/assert`)

Reply:
I just wanted to mention `require := require.New(t)` for longer test cases to clean up the arguments to future require calls.

Reply:
done in https://github.com/rancher/gitjob/pull/287/commits/7ce3b9b02d21dfc1295448d079d6041d80249880